src/test/java/com/williamfiset/algorithms/graphtheory/TravelingSalesmanProblemTest.java

```diff
@@ -11,58 +11,85 @@
 
   @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingRecursive(321, dist);
+    new TspDynamicProgrammingRecursive(321, new double[5][5]);
   }
 
   @Test(expected = IllegalArgumentException.class)
-  public void testTspIterativeInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingIterative(321, dist);
+  public void testTspIterativeStartNodeTooLarge() {
+    new TspDynamicProgrammingIterative(321, new double[5][5]);
   }
 
-  @Test(expected = IllegalStateException.class)
+  @Test(expected = IllegalArgumentException.class)
+  public void testTspIterativeNegativeStartNode() {
+    new TspDynamicProgrammingIterative(-1, new double[5][5]);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveNonSquareMatrix() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6}
-    };
-    new TspDynamicProgrammingRecursive(dist);
+    new TspDynamicProgrammingRecursive(new double[5][4]);
   }
 
   @Test(expected = IllegalStateException.class)
   public void testTspIterativeNonSquareMatrix() {
-    double[][] dist = {
```

Comment:
For example, this test didn't test non-square matrices because the dist matrix is 2x3 and this gets caught by the handler for matrices with N <= 2 instead.